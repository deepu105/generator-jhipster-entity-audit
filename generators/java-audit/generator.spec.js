import { beforeAll, describe, expect, it } from 'vitest';

import { defaultHelpers as helpers, result } from 'generator-jhipster/testing';

const SUB_GENERATOR = 'java-audit';
const SUB_GENERATOR_NAMESPACE = `jhipster-entity-audit:${SUB_GENERATOR}`;

describe('SubGenerator java-audit of entity-audit JHipster blueprint', () => {
  for (const auditFramework of [undefined, 'javers', 'custom']) {
    describe(`run using auditFramework ${auditFramework}`, () => {
      beforeAll(async function () {
        await helpers
          .run(SUB_GENERATOR_NAMESPACE)
          .withJHipsterConfig({}, [
            {
              name: 'Audited',
              enableAudit: true,
              fields: [
                {
                  fieldName: 'name',
                  fieldType: 'String',
                },
              ],
            },
          ])
          .withOptions({
            creationTimestamp: '2022-01-01',
            ignoreNeedlesError: true,
            auditFramework,
          })
          .withJHipsterLookup()
          .withParentBlueprintLookup();
      });

      it('should succeed', () => {
        expect(result.getStateSnapshot()).toMatchSnapshot();
      });
      if (auditFramework) {
        it('entities should extend AbstractAuditingEntity', () => {
          result.assertFileContent('src/main/java/com/mycompany/myapp/domain/Audited.java', ' AbstractAuditingEntity<');
        });
      } else {
        it('entities should not extend AbstractAuditingEntity', () => {
          result.assertNoFileContent('src/main/java/com/mycompany/myapp/domain/Audited.java', ' AbstractAuditingEntity<');
        });
      }
    });
  }
});
