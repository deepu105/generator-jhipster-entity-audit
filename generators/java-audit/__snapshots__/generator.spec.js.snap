// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubGenerator java-audit of entity-audit JHipster blueprint > run > should succeed 1`] = `
{
  ".jhipster/Audited.json": {
    "stateCleared": "modified",
  },
  ".yo-rc.json": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/GeneratedByJHipster.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/Audited.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/domain/package-info.java": {
    "stateCleared": "modified",
  },
  "src/main/java/com/mycompany/myapp/package-info.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuditedTest.java": {
    "stateCleared": "modified",
  },
  "src/test/java/com/mycompany/myapp/domain/AuditedTestSamples.java": {
    "stateCleared": "modified",
  },
}
`;
